feat: pass filename as argument to print_by_server

KrithiRamani opened the pull request:
To print attached files(pdf or others) from click of a button in the doctype, we need to pass the filename. So added additional argument "file".  Hash will be generated if file is None.


`no-docs`

KrithiRamani committed:
passed filename as argument

KrithiRamani committed:
renamed argument to file_path. Added pycups to requirements.txt

ankush commented on frappe/utils/print_format.py:
```diff
@@ -98,7 +98,7 @@ def report_to_pdf(html, orientation="Landscape"):
 
 @frappe.whitelist()
-def print_by_server(doctype, name, printer_setting, print_format=None, doc=None, file=None,no_letterhead=0):
+def print_by_server(doctype, name, printer_setting, print_format=None, doc=None, file_path=None,no_letterhead=0):
```
```suggestion
def print_by_server(doctype, name, printer_setting, print_format=None, doc=None, no_letterhead=0, file_path=None):
```

new args should be at the end for supporting non-kwarg function calls.

ankush commented on requirements.txt:
```diff
@@ -77,3 +77,4 @@ wrapt~=1.12.1
 xlrd~=2.0.1
 zxcvbn-python~=4.4.24
 tenacity~=8.0.1
+pycups
```
```suggestion
pycups~=2.0.1
```

KrithiRamani committed:
Update frappe/utils/print_format.py

As suggested, moved new args to the end for supporting non-kwarg function calls.

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>

KrithiRamani committed:
Update requirements.txt

Added version no for pycups

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>

rohitwaghchaure commented:
@Mergifyio backport version-13-hotfix

ankush commented:
Just realized why pycups was previously left out of requirements.txt

It requires `libcups2-dev` devel libraries while installing, so unless the user has that lib, installation of Frappe just fails. So this change adds another system-level dependency that's not documented in our installation process. 

docs for network printing: https://docs.erpnext.com/docs/v13/user/manual/en/setting-up/print/print-settings#3-network-printer-print-server 

We can either:
1. Remove pycups
2. Add the system dependency in the installation guide. `libcups2-dev`

Frappe CI isn't breaking right now because in `install.sh` we are installing this library (not sure why 🤷 , it's not getting tested anyway) but other apps are breaking: https://github.com/frappe/ecommerce_integrations/runs/3765418884?check_suite_focus=true

@rohitwaghchaure @KrithiRamani @gavindsouza
